### DISCUSSION TITLE
Using Rust Bitcoin to Access bitcoind chain data directly

### DISCUSSION THREAD

**Discussion by EthnTuttle** [rust-bitcoin/rust-bitcoin#2995] [Show and tell]
https://github.com/satonomics-org/satonomics

This isn't my project but I found it interesting that it stops the bitcoind instance and references the underlying bitcoin data files directly. This is contrasting with the usual use of RPC calls.

**Comment by apoelstra** [comment]
I'm not sure if you're suggesting it, but FYI we are not going to support this. People have been doing this for over a decade and it has been annoying Bitcoin Core developers just as long. The format/location/distribution of these files has no stability guarantee, and if you write to them while they are open then bad things will happen (though I see that this project shuts down the node). You should only be doing this to do forensics on your node. Though I do understand that it's much faster to just read the .dat files, so no matter how much I complain, that's what the incentives are.

**Reply by Kixunil** [reply]
For a while I was thinking of having a service external to Core that would store the blocks in its own stable-ish format (pruning Core) and provide a dylib so that one can read them even if the format changes. This would make electrs faster among other things.

Or maybe if Core RPC was much faster the incentive would no longer be there. :P
